src/Storages/VirtualColumnUtils.cpp

```diff
@@ -170,13 +170,13 @@ VirtualColumnsDescription getVirtualsForFileLikeStorage(ColumnsDescription & sto
 {
     VirtualColumnsDescription desc;
 
-    auto add_virtual = [&](const NameAndTypePair & pair)
+    auto add_virtual = [&](const NameAndTypePair & pair, bool prefer_virtual_column) /// By using prefer_virtual_column we define whether we will overwrite the storage column with the virtual one
     {
         const auto & name = pair.getNameInStorage();
         const auto & type = pair.getTypeInStorage();
         if (storage_columns.has(name))
         {
-            if (!context->getSettingsRef()[Setting::use_hive_partitioning])
+            if (prefer_virtual_column)
```

Comment:
Shouldn't it be `!prefer_virtual_column`?

Reply:
Yes